apps/dashboard/app/apps/nav_bar.rb

```diff
@@ -0,0 +1,109 @@
+class NavBar
+
+  attr_reader :nav_items
+
+  def self.from_config(nav_bar)
+    nav_items = nav_bar.map do |nav_item|
+      NavBar.parse_nav_item(nav_item)
+    end
+
+    NavBar.new(nav_items: nav_items)
+  end
+
+  def initialize(nav_items: [])
+    @nav_items = nav_items
+  end
+
+  def has_items?
+    nav_items.size > 0
+  end
+
+  def to_s
+    "nav_items: #{nav_items}"
+  end
+
+  private
+
+  def self.parse_nav_item(nav_item)
+    type = nav_item.fetch(:type, nil) || NavBar.infer_type(nav_item)
+    NavItem.new(type.to_sym, nav_item)
+  end
+
+  def self.infer_type(nav_item)
+    links = nav_item.fetch(:links, nil)
+    return :nav_menu if links
+
+    url = nav_item.fetch(:url, nil)
+    return :nav_link if url
+
+    token = nav_item.fetch(:token, nil)
+    return :nav_app if token
+
+    tokens = nav_item.fetch(:tokens, nil)
+    return :nav_app if tokens
+
+    template = nav_item.fetch(:template, nil)
+    return :nav_template if template
+
+    #default is divider
+    return :nav_divider
+  end
+
+  class NavItem
+    attr_reader :type, :items, :links, :title, :url, :icon, :new_tab
+
+    def initialize(type, config)
+      config = config.to_h.compact.symbolize_keys
+
+      @type      = type
+      @items     = config.fetch(:links, []).map{|link_item| NavBar.parse_nav_item(link_item)}
+
+      @title     = config.fetch(:title, nil)
+      @url       = config.fetch(:url, "").to_s
+      @icon      = URI(config.fetch(:icon, "fas://cog").to_s)
+      @new_tab   = !!config.fetch(:new_tab, true)
+
+      @token     = config.fetch(:token, nil)
+      @tokens    = config.fetch(:tokens, nil)
+      @template  = config.fetch(:template, nil)
+
+      @title = I18n.t(@title.delete_prefix("i18n://")) if @title && @title.start_with?("i18n://")
+
+      @links =  OodApp.new(Router.router_from_token(@token)).links if @token
+      if @tokens
+        apps = NavBar.filter_by_tokens(@tokens, NavBar.flat_sys_apps)
+        @links = apps.each_with_object([]) do |app, links|
+          links.concat(app.respond_to?(:links) ? app.links : [app.link])
+        end
+      end
+    end
+
+    def items?
+      items.size > 0
+    end
+
+    def template
+      @template ? "layouts/nav/#{@template}" : "layouts/nav/custom/#{type}"
+    end
+  end
+
+  def self.flat_sys_apps
+    SysRouter.apps.each_with_object([]) do |app, flat_list|
+      if app.has_sub_apps?
+        flat_list.concat(app.sub_app_list.map{|sub_app| FeaturedApp.new(sub_app.router, token: sub_app.token)})
+      else
+        flat_list.append(FeaturedApp.new(app.router, token: app.token))
+      end
```

Comment:
Using `FeaturedApp` here to have a common interface for apps and sub apps.

Before I was returning an array with a mixture of `OodApp` and `BatchConnect::App`